network/peers/peers_index.go

```diff
@@ -110,27 +112,41 @@ func (pi *peersIndex) UpdateSelfRecord(newSelf *records.NodeInfo) {
 	pi.selfLock.Lock()
 	defer pi.selfLock.Unlock()
 
-	pi.selfSealed = nil
 	pi.self = newSelf
 }
 
 func (pi *peersIndex) Self() *records.NodeInfo {
 	return pi.self
 }
 
-func (pi *peersIndex) SelfSealed() ([]byte, error) {
+func (pi *peersIndex) SelfSealed(sender, recipient peer.ID, operatorPrivateKey *rsa.PrivateKey) ([]byte, error) {
 	pi.selfLock.Lock()
 	defer pi.selfLock.Unlock()
 
-	if len(pi.selfSealed) == 0 {
-		sealed, err := pi.self.Seal(pi.netKeyProvider())
-		if err != nil {
-			return nil, err
-		}
-		pi.selfSealed = sealed
+	publicKey, err := rsaencryption.ExtractPublicKeyPem(operatorPrivateKey)
+	if err != nil {
+		return nil, err
+	}
+
+	handshakeData := records.HandshakeData{
+		SenderPeerID:    sender,
+		RecipientPeerID: recipient,
+		Timestamp:       time.Now(),
+		SenderPubKeyPem: publicKey,
+	}
+	hash := handshakeData.Hash()
+
+	signature, err := rsa.SignPKCS1v15(rand.Reader, operatorPrivateKey, crypto.SHA256, hash[:])
```

Comment:
rand isn't used so you could probably pass nil there

Reply:
it used here `/usr/local/go/src/crypto/rsa/rsa.go:543`

Reply:
figure it out, thanks
changed